stages/test/test_skopeo.py

```diff
@@ -33,3 +42,95 @@ def test_schema_validation_skopeo(stage_schema, test_data, expected_err):
     else:
         assert res.valid is False
         testutil.assert_jsonschema_error_contains(res, expected_err, expected_num_errs=1)
+
+
+@pytest.mark.skipif(os.getuid() != 0, reason="needs root")
+@pytest.mark.skipif(not has_executable("podman"), reason="no podman executable")
+@pytest.mark.parametrize("dest_type,local_output_path", [
+    ("dir", "/tmp/test-output-skopeo-dir"),
+    ("oci-archive", "/tmp/test-output-skopeo.tar"),
+    ("oci", "/tmp/test-output-skopeo-dir"),
+])
+def test_skopeo_dir(tmp_path, stage_module, dest_type, local_output_path):
+    with make_container(tmp_path, {"file1": "file1 from base"}) as base_tag:
+        with make_container(tmp_path, {"file1": "file1 from final layer"}, base_tag) as cont_tag:
+            # export for the container-deploy stage
+            fake_container_dst = tmp_path / "fake-container"
+            subprocess.check_call([
+                "podman", "save",
+                "--format=oci-archive",
+                f"--output={fake_container_dst}",
+                cont_tag,
+            ])
+
+    inputs = {
+        "images": {
+            # seems to be unused with fake_container_path?
+            "path": fake_container_dst,
+            "data": {
+                "archives": {
+                    fake_container_dst: {
+                        "format": "oci-archive",
+                        "name": cont_tag,
+                    },
+                },
+            },
+        },
+    }
+
+    output_dir = tmp_path / "output"
+    options = {
+        "destination": {
+            "type": dest_type,
+            "path": local_output_path
+        }
+    }
+
+    stage_module.main(inputs, output_dir, options)
+
+    assert output_dir.exists()
+
+    result = output_dir / local_output_path.strip("/")
+    assert_file_exists(result)
+
+    # Debugging in case of assertion failed
+    print("Output dir contains:")
```

Comment:
I would prefer if we avoid this and think about alternatives instead. The reason is that this tends to make tests very chatty and while pytest hides the output by default the output will still be there when something does fail or the test needs adjusting and someone runs it with `-s`. So here we could have a helper "listdir()" and put it next to the assert so that it only runs if the test fails. Also pytest leaves the tmp dirs for failed tests so one can inspect, e.g.
```
f = PosixPath('/tmp/pytest-of-root/pytest-23/test_skopeo_dir_dir__dstdir_te0/output/dstdir/test-output-skopeo-dir/versionx')

    def assert_file_exists(f: pathlib.Path):
>       assert f.exists(), f"'{str(f)}' not found"
E       AssertionError: '/tmp/pytest-of-root/pytest-23/test_skopeo_dir_dir__dstdir_te0/output/dstdir/test-output-skopeo-dir/versionx' not found
```

Reply:
> Also pytest leaves the tmp dirs for failed tests so one can inspect

oh wow, TIL :laughing: